include/sparse_bundle_adjustment/proj.h

```diff
@@ -49,7 +49,7 @@ namespace sba
   
   /// Obnoxiously long type def for the map type that holds the point 
   /// projections in tracks.
-  typedef std::map<const int, Proj, std::less<int>, Eigen::aligned_allocator<Proj> > ProjMap;
+  typedef std::map<const int, Proj, std::less<int>, Eigen::aligned_allocator<std::pair<const int, Proj>> > ProjMap;
```

Comment:
I don't understand - why are you making this into a pair?

Reply:
I followed e.g. https://github.com/JakobEngel/dso/pull/113/files - if I don't make it as a pair, it fails to build; I will provide the compile error later.